hadoop-ozone/recon/src/main/resources/webapps/recon/ozone-recon-web/src/v2/components/select/singleSelect.tsx

```diff
@@ -0,0 +1,78 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React from "react";
+import Select, {
+  Props as ReactSelectProps,
+  components,
+  ValueType,
+  ValueContainerProps,
+  StylesConfig
+} from 'react-select';
+
+import { selectStyles } from "@/v2/constants/select.constants";
+
+
+export type Option = {
+  label: string;
+  value: string;
+}
+
+interface SingleSelectProps extends ReactSelectProps<Option, false> {
+  options: Option[];
+  placeholder: string;
+  onChange: (arg0: ValueType<Option, false>) => void;
+}
+
+const SingleSelect: React.FC<SingleSelectProps> = ({
+  options = [],
+  placeholder = 'Limit',
+  onChange = () => { },
+  ...props
+}) => {
+
+
+  const ValueContainer = ({ children, ...props }: ValueContainerProps<Option, false>) => {
+    const selectedLimit = props.getValue() as Option[]
+    return (
+      <components.ValueContainer {...props}>
+        Limit: {selectedLimit[0].label}
```

Comment:
The `selectedLimit[0].label `logic assumes that `selectedLimit` always has at least one selected option. This could lead to errors if `selectedLimit` is empty. Consider adding a conditional check.
```suggestion
        const selectedLimit = props.getValue() as Option[];
const label = selectedLimit.length > 0 ? selectedLimit[0].label : '';
return (
  <components.ValueContainer {...props}>
    Limit: {label}
  </components.ValueContainer>
);

```

Reply:
Fixed in the latest commit [1347224](https://github.com/apache/ozone/pull/7048/commits/13472245263b08e28aeda28b1f3597df5abc3396).

Made it as
```
<components.ValueContainer {...props}>
    Limit: {selectedLimit[0]?.label ?? ''}
</components.ValueContainer>
```

This will automatically assign `''` as the default value in case the limit is undefined